### DISCUSSION TITLE
Is there contuious integration reports? Just to know that commits pass the build,  the tests, etc.

### DISCUSSION THREAD

**Discussion by belonesox** [sagemathinc/cocalc#7004] [Q&A]
As I know, cocalc do not use releases/tags/... to somehow mark commits that 
* can be deployed, 
* may be even cocalc-docker starts works and pass smoke tests ("can start project"), 
* or even can be be builded by `pnpm make` without errors.
 
Is there contuious integration reports? (Looks like CI exists)

Just to know that commits pass the build,  the tests (what tests), etc. ...

**Comment by haraldschilly** [comment]
You can look into [Actions](https://github.com/sagemathinc/cocalc/actions), which run builds and tests. Their result is visible next commits as a :heavy_check_mark: or :x: ... you can even click on these icons.

![Screenshot from 2023-10-25 14-24-06](https://github.com/sagemathinc/cocalc/assets/207405/76813e54-dc0f-4610-bea1-069c21eaad5f)

**Reply by williamstein** [reply]
Also, what Actions is doing is just the same as running `pnpm test` for all packages.   You should also be able to do that locally, assuming your database is setup.  Testing of some components uses a live local postgresql server.

**Comment by belonesox** [comment]
Unfortunately with github actions I cannot easily get on what exact commit they are running ... 
same branch, same pull request — and they can fail or pass 

![image](https://github.com/sagemathinc/cocalc/assets/1609739/ec1e5646-2c34-4c63-8f25-51764793f29d)

«master» branch also can be broken (on my experience), and I do not find something like «production»  branch («cocalc-docker-updates»?).

May be automatically publish somewhere  commit version of cocalc-docker image? 
(to get it without pulling/running docker image and digging inside)
